plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts

```diff
@@ -313,6 +316,25 @@ export function splitIngestionBatch(
         for (const message of kafkaMessages) {
             // Drop based on a token blocklist
             const pluginEvent = formPipelineEvent(message)
+
+            // We de-duplicate merge events here, because if we don't, the later concurrency will cause
+            // us to lock up PG, and if we're getting a large amount of duplicated merge events, bring down
+            // the whole pipeline.
+            if (['$create_alias', '$merge_dangerously', '$identify', '$anon_distinct_id'].includes(pluginEvent.event)) {
+                const alias = pluginEvent.properties?.alias || "I don't know what should be here"
```

Comment:
Not sure this works... I would need to look into it a little more but I think `$identify` for example has a different property for the aliasing.

Reply:
I think it would need to be done deeper at the actual level of doing the merges